src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,87 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
 public class StringManipulation implements StringManipulationInterface {
 
+    String s;
     @Override
     public String getString() {
-        return null;
+        return s;
     }
 
     @Override
     public void setString(String string) {
+        s = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if (s == null || s.isEmpty())
+            return 0;
+        return s.split("\\s+").length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        if (n <= 0) {
+            throw new IllegalArgumentException();
```

Comment:
It may be helpful to provide am message pertaining to the exact reason illegalArgumentException is called. This may help clarify during testing to save some time.